cmd/callhome.go

```diff
@@ -118,26 +108,85 @@ func runCallhome(ctx context.Context, objAPI ObjectLayer) bool {
 }
 
 func performCallhome(ctx context.Context) {
-	err := sendCallhomeInfo(
-		CallhomeInfo{
-			SchemaVersion: callhomeSchemaVersion,
-			AdminInfo:     getServerInfo(ctx, nil),
-		})
-	if err != nil {
-		logger.LogIf(ctx, fmt.Errorf("Unable to perform callhome: %w", err))
+	deadline := 10 * time.Second // Default deadline is 10secs for callhome
+	objectAPI := newObjectLayerFn()
+
+	healthCtx, healthCancel := context.WithTimeout(ctx, deadline)
+	defer healthCancel()
+
+	healthInfoCh := make(chan madmin.HealthInfo)
+
+	query := url.Values{}
+	for _, k := range madmin.HealthDataTypesList {
+		query[string(k)] = []string{"true"}
+	}
+
+	healthInfo := madmin.HealthInfo{
+		Version: madmin.HealthInfoVersion,
+		Minio: madmin.MinioHealthInfo{
+			Info: madmin.MinioInfo{
+				DeploymentID: globalDeploymentID,
+			},
+		},
+	}
+
+	go fetchHealthInfo(healthCtx, objectAPI, &query, healthInfoCh, healthInfo)
+
+	for {
+		select {
+		case hi, ok := <-healthInfoCh:
+			if ok {
+				healthInfo = hi
+			}
+		case <-healthCtx.Done():
+			err := sendHealthInfo(ctx, healthInfo)
```

Comment:
so we are only sending health only after a context is done? logic must be IMO sendHealthInfo() when channel returns and do a blanket return here is when context is done. 

Relying on context done and then sending a global variable I don't know if its it very useful. IS this done to ensure that if it takes too long we still send some partial data?

Reply:
@harshavardhana the channel returns multiple times with progressively more data, the last one containing all the data of the health info. I'm trying to make a single API call to SUBNET with the full health info data in the payload.

Reply:
Thanks, fixed.